mleap-core/src/main/scala/ml/combust/mleap/core/feature/StringIndexerModel.scala

```diff
@@ -16,28 +16,29 @@ case class StringIndexerModel(labels: Seq[String],
                               handleInvalid: HandleInvalid = HandleInvalid.Error) extends Model {
   val stringToIndex: Map[String, Int] = labels.zipWithIndex.toMap
   private val keepInvalid = handleInvalid == HandleInvalid.Keep
-
+  private val invalidValue = labels.length
   /** Convert a string into its integer representation.
     *
     * @param value label to index
     * @return index of label
     */
   def apply(value: Any): Int = if (value == null) {
     if (keepInvalid) {
-      labels.length
+      invalidValue
     } else {
       throw new NullPointerException("StringIndexer encountered NULL value. " +
         s"To handle NULLS, set handleInvalid to ${HandleInvalid.Keep.asParamString}")
     }
   } else {
     val label = value.toString
-    if (stringToIndex.contains(label)) {
```

Comment:
If `contains` and `apply` are both O(1), does it really matter that we do both every time?

Reply:
HashingMap can be O(1), But not always. 
And if it's TreeMap, then It's O(Log(N)).

https://alvinalexander.com/scala/understanding-performance-scala-collections-classes-methods-cookbook/